Change detector fails when adding new test package

e.g.
```
=== FAIL: tests/integration/client_introspection  (0.00s)
Cloning into '/tmp/defradb/tests/changeDetector/code/79af7410c96dbd93813ef07f55a7b22efc40e513'...
2023-05-11T19:54:11.352Z, ERROR, defra.tests.integration, , {"Error": "chdir /tmp/defradb/tests/changeDetector/code/79af7410c96dbd93813ef07f55a7b22efc40e513/tests/integration/client_introspection: no such file or directory"}
panic: chdir /tmp/defradb/tests/changeDetector/code/79af7410c96dbd93813ef07f55a7b22efc40e513/tests/integration/client_introspection: no such file or directory

goroutine 1 [running]:
github.com/sourcenetwork/defradb/tests/integration.detectDbChangesInit({0x1941f47, 0x2c}, {0x17c3797, 0x7})
        /home/o/defradb/tests/integration/change_detector.go:158 +0x990
github.com/sourcenetwork/defradb/tests/integration.init.0()
        /home/o/defradb/tests/integration/utils2.go:133 +0x2f1
FAIL    github.com/sourcenetwork/defradb/tests/integration/client_introspection 1.647s
```

--- a/tests/integration/change_detector.go
+++ b/tests/integration/change_detector.go
@@ -134,6 +134,16 @@ func detectDbChangesInit(repository string, targetBranch string) {
 
 	targetTestPackage := detectDbChangesCodeDir + "/tests/integration/" + testPackagePath
 
+	_, err = os.Stat(targetTestPackage)
+	if os.IsNotExist(err) {
+		// This is a new test package, and thus the change detector is not applicable
+		// as the tests do not exist in the target branch.
+		skip = true
+		return
+	} else if err != nil {
+		panic(err)
+	}
+
 	// If we are checking for database changes, and we are not seting up the database,
 	// then we must be in the main test process, and need to create a new process
 	// setting up the database for this test using the old branch We should not setup

